example/ssh2_exec.c

```diff
@@ -262,7 +262,7 @@ int main(int argc, char *argv[])
                     fprintf(stderr, "libssh2_channel_read returned %ld\n",
                             (long)nread);
             }
-        } while(nread > 0);
+        } while(nread > 0 || nread == LIBSSH2_ERROR_EAGAIN);
```

Comment:
On a second look: This change makes the loop continue on `EAGAIN`. However the outer loop expects this value and the inner loop to exit on that value. On `EAGAIN` it calls `waitsocket()` and enters the inner loop again, otherwise exists.

Am I overlooking something?